docs/deployment/setup_gardener.md

```diff
@@ -1,44 +1,412 @@
+- [Accessing the virtual Garden](https://github.com/gardener/gardener/blob/master/docs/concepts/operator.md#virtual-garden-kubeconfig)
+
+### Extensions
+
+Gardener is provider-agnostic and relies on a growing set of extensions. An extension can provide support for an infrastructure (required) or add a desired feature to a Kubernetes cluster (often optional).
+
+There are a few extensions considered essential to any Gardener installation:
+
+- at least one [infrastructure provider](https://github.com/gardener/gardener/tree/master/extensions#infrastructure-provider)
+- at least one [operating system extension](https://github.com/gardener/gardener/tree/master/extensions#operating-system) - [Garden Linux](https://github.com/gardener/gardener-extension-os-gardenlinux) is recommended
+- at least one [network plugin (CNI)](https://github.com/gardener/gardener/tree/master/extensions#network-plugin)
+- Shoot services
+
+An extension often consists of two components - a controller implementing the Gardener's extension contract and an admission controller. The latter is deployed to the runtime cluster and prevents misconfiguration of `Shoots`.
+
+#### Choices
+
+For this example, the following choices are made:
+- [Garden Linux OS extension](https://github.com/gardener/gardener-extension-os-gardenlinux)
+- [Provider Openstack](https://github.com/gardener/gardener-extension-provider-openstack)
+- [Calico](https://github.com/gardener/gardener-extension-networking-calico)
+- [DNS service](https://github.com/gardener/gardener-extension-shoot-dns-service)
+- [Cert service](https://github.com/gardener/gardener-extension-shoot-cert-service)
+- [OIDC service](https://github.com/gardener/gardener-extension-shoot-oidc-service)
+
+Reference documentation:
+- [Extension Contract](https://github.com/gardener/gardener/blob/master/docs/extensions/resources/extension.md#contract-extension-resource)
+- [GEP 01 Extensibility](https://github.com/gardener/gardener/blob/master/docs/proposals/01-extensibility.md)
+- [Extension Library](https://github.com/gardener/gardener/tree/master/extensions#known-extension-implementations)
+
+#### How to install an extension
+
+In order to make an extension known to a Gardener landscape, two resources have to be applied to the virtual Garden cluster - firstly, a `ControllerDeployment` and secondly, a matching `ControllerRegistration`. With both in place, Gardener-managed `ControllerInstallations` take care of the actual deployment of the extension to the target environment.
+
+Check the [controller registration documentation](https://github.com/gardener/gardener/blob/master/docs/extensions/controllerregistration.md) for more details.
+
+Now, the classical way of installing `ControllerDeployment` and `ControllerRegistration` is to prepare the resources and apply them to the virtual Garden cluster. Please note, the `ControllerDeployment` contains a base64 encoded archive of extension's helm chart.
```

Comment:
> Please note, the `ControllerDeployment` contains a base64 encoded archive of extension's helm chart

Since v1 version of ControllerDeployment it is possible to use helm charts stored in oci repository, and this was made backward compatible with v1beta1. See https://github.com/gardener/gardener/blob/master/example/25-controllerdeployment.yaml

Reply:
Adapted with a37e8e4b077f2b6ae111606124d82f0448377565